cyral/resource_cyral_repository.go

```diff
@@ -39,138 +55,382 @@ func repositoryTypes() []string {
 	}
 }
 
+func mongoServerTypes() []string {
+	return []string{
+		"replicaset",
+		"standalone",
+	}
+}
+
+type RepoInfo struct {
+	ID                       string           `json:"id"`
+	Name                     string           `json:"name"`
+	Type                     string           `json:"type"`
+	Host                     string           `json:"repoHost"`
+	Port                     uint32           `json:"repoPort"`
+	ConnParams               *ConnParams      `json:"connParams"`
+	Labels                   []string         `json:"labels"`
+	RepoNodes                []*RepoNode      `json:"repoNodes,omitempty"`
```

Comment:
I prefer to use pointers as it makes checking unset values quicker (rather than checking if the value is the default go value, for each field in a struct). For example, if the preferredAccessGateway is unset, without pointers, I have to check if both the bindingID and sidecarID are empty strings.

Reply:
I agree with the use of pointers for the reasons you mentioned, I was just trying to understand why the RepoNodes needed to be a slice of pointers instead of just `[]RepoNodes`, if the API allows nil RepoNode items in the list then using `[]*RepoNode` makes sense.